elements/rh-site-status/rh-site-status.ts

```diff
@@ -0,0 +1,149 @@
+  `
+};
+
+// map statuspage.io's text to our text; at least one of their status
+// strings is too long for the space we have
+const textMap: Record<string, string> = {
+  'Partially Degraded Service': 'Partial service',
+};
+
+// map statuspage.io's statuses to icon map keys
+const statusMap: Record<string, string> = {
+  'maintenance': 'warn',
+  'critical': 'danger',
+  'major': 'warn',
+  'minor': 'warn',
+  'none': 'ok',
+};
+
+interface ApiStatus {
+  status: {
+    description: string;
+    indicator: string;
+  };
+}
+
+/**
+ * Website status communicates the operational status of a website or domain using a status icon and link. It is usually located in the Footer component.
+ *
+ * @summary Communicates operational status of a website or domain
+ */
+@customElement('rh-site-status')
+export class RhSiteStatus extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Sets color context for child components, overrides parent context
+   */
+  @colorContextProvider()
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette = 'dark';
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  #logger = new Logger(this);
+
+  #text = 'Loading';
+
+  #icon: TemplateResult = html``;
+
+  #isLoading = true;
+
+  #isApiStatus = (data: ApiStatus): data is ApiStatus => {
+    return (
+      typeof data === 'object' &&
+      data !== null &&
+      'status' in data &&
+      data.status !== null &&
+      'description' in data.status &&
+      'indicator' in data.status
+    );
+  };
+
+  async connectedCallback() {
+    super.connectedCallback();
+    await this.#getStatus();
+  }
+
+  render() {
+    const { on = '' } = this;
+    return html`
+      <a href="https://status.redhat.com/" part="container" class="${classMap({ [on]: !!on })}">
+        ${this.#isLoading ? html`<rh-spinner size="sm"></rh-spinner>` : html`
+          <svg xmlns="http://www.w3.org/2000/svg" width="16" height="17" viewBox="0 0 16 17" fill="none">
+            ${this.#icon}
+          </svg>
+        `}
+        ${this.#text}
+      </a>
+    `;
+  }
+
+  async #getStatus() {
+    await fetch('https://status.redhat.com/index.json', {
+      mode: 'cors',
+      cache: 'no-cache',
+      headers: {
+        Accept: 'application/json',
+      }
+    })
+      .then(response => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error(`${response.status} ${response.statusText}`);
+        }
+      })
+      .then((data: ApiStatus) => {
+        if (!this.#isApiStatus(data)) {
+          throw new Error('Invalid status data');
```

Comment:
Elements shouldn't throw just by connecting

So handle this error somewhere

Reply:
After trying to grok this comment, I reverted the change I started to make.  

Can you please give me a bit more detail on what you are expecting here?  

The throw is there in order to catch the 404 or other network errors that returns if the API URL is incorrect/ service is down etc.  That is handled by the catch in the promise chain.  

I feel like doing this elsewhere would imho create extra state in the component which just feels extraneous.  Also I was thinking you might have meant to remove it from the connectedCallback lifecycle, but moving somewhere else would then cause a double render as well correct?

If I'm off base or missing something let me know.